### PR TITLE
fix(cli): region specified in `~/.aws/credentials` is ignored

### LINKED ISSUE #32130: CDK CLI: region must be in `config` file (ignored in `credentials` file) since 2.167.0
### Describe the bug

From this thread: https://github.com/aws/aws-cdk/issues/32120

By @otbe:

----------

For us the issue is in the region, e.g.

deploy -vvv
```
10:38:42] Looking up AWS region in the EC2 Instance Metadata Service (IMDS).
[10:38:43] Unable to retrieve AWS region from IMDS: Error: Error fetching metadata token: TimeoutError: Connection timed out after 1000 ms
[10:38:43] Unable to determine AWS region from environment or AWS configuration (profile: "default"), defaulting to 'us-east-1'
```

Not quite sure how it worked in older aws-cdk versions but the region is specified in `.aws/credentials` similar to this

```
[default]
aws_access_key_id=
aws_secret_access_key=
aws_session_token=
aws_security_token=
region=eu-central-1
```

That means our stacks are deployed in the wrong region with 2.167.0 🫠
Reverting back to 2.166.0 solves the issue.

### Regression Issue

- [X] Select this option if this issue appears to be a regression.

### Last Known Working CDK Version

_No response_

### Expected Behavior

Uses my configured region in the INI file

### Current Behavior

Uses `us-east-1`

### Reproduction Steps

See description

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.167.0

### Framework Version

_No response_

### Node.js Version

-

### OS

-

### Language

TypeScript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by iliapolo**
We just didn't consider the shared credentials file as returned by the SDK when loading configuration. 

Closes #32130.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts [hunk 1]
```diff
@@ -149,7 +149,12 @@ export class AwsCliCompatible {
  */
 async function getRegionFromIni(profile: string): Promise<string | undefined> {
   const sharedFiles = await loadSharedConfigFiles({ ignoreCache: true });
-  return sharedFiles?.configFile?.[profile]?.region || sharedFiles?.configFile?.default?.region;
+  // https://docs.aws.amazon.com/cli/v1/userguide/cli-configure-files.html
+  return getRegionFromIniFile(profile, sharedFiles.configFile) ?? getRegionFromIniFile(profile, sharedFiles.credentialsFile);
```
Comment (iliapolo) [inline packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts:153]: I'm not sure what takes precedence: `config` or `credentials`? Looking it up but if anybody knows do tell.

Note that in `config` file, profiles are suffixed with the `profile ` string. Nonetheless, you the SDK parses it, it ends having the same keys as the `credentials` file, so there's no need to add the prefix when querying the file.

```console
ConfigFile: {
  "default": {
    "region": "us-east-2"
  },
  "user1": {
    "region": "us-east-2"
  }
}
CredentialsFile: {
  "default": {
    "region": "us-east-2"
  },
  "user1": {
    "region": "us-east-2"
  }
}
```

For files that look like: 

```console
❯ cat ~/.aws/config                                                                                                                                                                                                                             [13:52:11]
[default]
region=us-east-2

[profile user1]
region=us-east-2

 ❯ cat ~/.aws/credentials                                                                                                                                                                                                                        [13:52:16]
[default]
region=us-east-2

[user1]
region=us-east-2
```

Comment (rix0rrr) [reply packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts:153]: I was thinking of the order:

- Profile config file
- Profile credentials file
- Default config file
- Default credentials file

So profile strictly before default.

Thoughts?

**Comment by rix0rrr** [comment]
AWS CLI does:

* If `--profile`: credentials profile > config profile > no region
* If not `--profile`: credentials default > config default > no region

No region behaves as follows:
* Global services work (IAM, S3, Route53)
* Regional services fail with an error, saying you must specify a region

Traditionally, CDK has defaulted to us-east-1 in this case. I'm like 90% sure we used to do that to copy the AWS CLI, I wonder if they changed their behavior when migrating CLIv1 -> CLIv2

Also CDK traditionally had the behavior that if the profile doesn't have a region, we will take the region from the [default] profile, but apparently the CLI doesn't do that.

**Comment by iliapolo** [comment]
draft until we sort our credentials priority

File: packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts [hunk 2]
```diff
@@ -149,7 +149,25 @@ export class AwsCliCompatible {
  */
 async function getRegionFromIni(profile: string): Promise<string | undefined> {
   const sharedFiles = await loadSharedConfigFiles({ ignoreCache: true });
-  return sharedFiles?.configFile?.[profile]?.region || sharedFiles?.configFile?.default?.region;
+
+  // Priority:
+  //
+  // credentials come before config because aws-cli v1 behaves like that.
```
Comment (rix0rrr) [inline packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts:155]: Not entirely true that AWS CLI v1 behaves this way, but we're ~consistent with how users will *probably* use it 😉

**Pull request merged**